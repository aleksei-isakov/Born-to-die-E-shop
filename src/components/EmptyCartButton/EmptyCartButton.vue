<template>
  <base-text-border-button
    color="custom-grey"
    class="empty-cart-button"
    @click="onClickShowPopup"
  >
    Empty cart
  </base-text-border-button>
</template>

<script>
import { BaseTextBorderButton } from '@/base_components/';

export default {
  name: 'EmptyCartButton',

  components: {
    BaseTextBorderButton
  },

  methods: {
    onClickShowPopup() {
      this.$emit('show-popup');
    }
  }
};
</script>
