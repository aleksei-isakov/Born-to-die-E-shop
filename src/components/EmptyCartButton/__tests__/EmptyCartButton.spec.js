import { shallowMount, createLocalVue } from '@vue/test-utils';
import EmptyCartButton from '../../EmptyCartButton/EmptyCartButton.vue';

describe('EmptyCartButton.vue', () => {
  let wrapper;

  beforeEach(() => {
    const localVue = createLocalVue();

    wrapper = shallowMount(EmptyCartButton, {
      localVue
    });

    expect(wrapper.find('button')).toBeTruthy();
    expect(wrapper.is(EmptyCartButton)).toBeTruthy();
  });

  it('renders a valid snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
