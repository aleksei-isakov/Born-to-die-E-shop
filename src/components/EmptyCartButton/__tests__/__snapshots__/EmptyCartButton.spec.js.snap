// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyCartButton.vue renders a valid snapshot 1`] = `
<base-text-border-button-stub
  class="empty-cart-button"
  color="custom-grey"
>
  
  Empty cart

</base-text-border-button-stub>
`;
